bot/cogs/news.py

```diff
@@ -0,0 +1,255 @@
+import logging
+import typing as t
+from datetime import date, datetime
+
+import discord
+import feedparser
+from bs4 import BeautifulSoup
+from dateutil import tz
+from discord.ext.commands import Cog
+from discord.ext.tasks import loop
+
+from bot import constants
+from bot.bot import Bot
+
+PEPS_RSS_URL = "https://www.python.org/dev/peps/peps.rss/"
+
+RECENT_THREADS_TEMPLATE = "https://mail.python.org/archives/list/{name}@python.org/recent-threads"
+THREAD_TEMPLATE_URL = "https://mail.python.org/archives/api/list/{name}@python.org/thread/{id}/"
+MAILMAN_PROFILE_URL = "https://mail.python.org/archives/users/{id}/"
+THREAD_URL = "https://mail.python.org/archives/list/{list}@python.org/thread/{id}/"
+
+AVATAR_URL = "https://www.python.org/static/opengraph-icon-200x200.png"
+
+log = logging.getLogger(__name__)
+
+
+class News(Cog):
+    """Post new PEPs and Python News to `#python-news`."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.webhook_names = {}
+        self.webhook: t.Optional[discord.Webhook] = None
+        self.channel: t.Optional[discord.TextChannel] = None
+
+        self.bot.loop.create_task(self.get_webhook_names())
+        self.bot.loop.create_task(self.get_webhook_and_channel())
+
+    async def start_tasks(self) -> None:
+        """Start the tasks for fetching new PEPs and mailing list messages."""
+        self.post_pep_news.start()
+        self.post_maillist_news.start()
```

Comment:
Should we have a cog unload method to stop these tasks? If the cog is reloaded presumably these tasks will be running twice?